### DISCUSSION TITLE
Change transition ?

### DISCUSSION THREAD

**Discussion by skrew** [davdroman/swiftui-navigation-transitions#73] [Q&A] [answered]
Hi,
I'm using it and it works fine, except 1 problem:

I want to use default transition for subview transition

What i do:

```
NavigationStack(path: $path) {
...
.navigationDestination(for: MediaItemInfo.self) { item in
    ...
}
.navigationStackTransition(
    .slide(axis: .vertical).animation(.linear(duration: 0.5))
)
```

Later, in the new view, i need the default transition for all subviews

```
NavigationLink(destination: ...) {
}
.navigationStackTransition(.default)
```

What is strange, is the transition are still the .slide one, but when i pop the new view, i have the standard transition.

What i'm missing ? Thanks

**Comment by davdroman** [comment, marked as answer]
I'm afraid you can't mix and match different transitions on the same `NavigationStack`. And even if you could, it's a bit of a UX anti-pattern anyway, so it might be worth considering an alternate way of organizing the navigation hierarchy in your app (e.g. by using modals).

As a side note, `navigationStackTransition` is deprecated in favor of `navigationTransition` in the last few versions of the library :)
